lib/srv/desktop/windows_server.go

```diff
@@ -243,8 +245,10 @@ func (cfg *WindowsServiceConfig) CheckAndSetDefaults() error {
 	if err := cfg.Heartbeat.CheckAndSetDefaults(); err != nil {
 		return trace.Wrap(err)
 	}
-	if err := cfg.LDAPConfig.Check(); err != nil {
-		return trace.Wrap(err)
+	if len(cfg.Heartbeat.StaticHosts) > 0 {
```

Comment:
This won't quite work, because it's completely valid to have 0 static hosts and discover them all via LDAP.

Maybe you can run `LDAPConfig.Check()` only if `LDAPConfig.Addr` is non-empty?

Reply:
I modified the code to run check if there are any static hosts or any DiscoveryBaseDN present. That should handle all case where LDAP is needed. 
Running check only `LDAPConfig.Addr` is non-empty would make it easy to accidentally leave it empty even if it's needed (too relaxed)